fiftyone/core/odm/database.py

```diff
@@ -124,7 +125,9 @@ def establish_db_conn(config):
 
             raise error
 
-    _client = pymongo.MongoClient(**_connection_kwargs)
+    _client = pymongo.MongoClient(
+        **_connection_kwargs, appname=foc.DATABASE_APPNAME
+    )
     _validate_db_version(config, _client)
```

Comment:
Perhaps start the external service after the DB version (and connection) are validated? Right now the service starts even for invalid URIs